serving/resource_model_serving.go

```diff
@@ -96,3 +137,31 @@ func ResourceModelServing() common.Resource {
 		},
 	}
 }
+
+func hasExactlyOneExternalModel(d *schema.ResourceDiff) error {
```

Comment:
I don't think this is what this is checking. It seems to verify two things:
1. that you specified the appropriate config for the provider of the first external model, and
2. that you did not specify any other configs for the first external model.
Let's correct the name here. Maybe `validateExternalModelConfig` is good?

BTW: it might be easier to write this check if you do
```go
var req serving.CreateServingEndpoint
common.DataToStructPointer(schema, d, &req)
```
and then do your checks on the typed `req` rather than trying to get the right fields from TF.

Reply:
Decided to go with the current approach since using a typed approach would require reflection and string mutation for CamelCase & snake_case